zrml/court/src/lib.rs

```diff
@@ -867,27 +882,47 @@ mod pallet {
             );
 
             let mut draws = <SelectedDraws<T>>::get(court_id);
-            match draws.binary_search_by_key(&who, |draw| draw.court_participant.clone()) {
-                Ok(index) => {
-                    let draw = draws[index].clone();
-
-                    let raw_commitment =
-                        RawCommitment { juror: who.clone(), vote_item: vote_item.clone(), salt };
-
-                    let commitment = Self::get_hashed_commitment(draw.vote, raw_commitment)?;
-
-                    let raw_vote =
-                        Vote::Revealed { commitment, vote_item: vote_item.clone(), salt };
-                    draws[index] = Draw { court_participant: who.clone(), vote: raw_vote, ..draw };
-                }
-                Err(_) => return Err(Error::<T>::CallerNotInSelectedDraws.into()),
-            }
+            let (slashable_amount, draw_weight) =
+                match draws.binary_search_by_key(&who, |draw| draw.court_participant.clone()) {
+                    Ok(index) => {
+                        let mut slashable_amount = BalanceOf::<T>::zero();
+                        let mut draw_weight = 0u32;
+                        if let Some(draw) = draws.get_mut(index) {
+                            let raw_commitment = RawCommitment {
+                                juror: who.clone(),
+                                vote_item: vote_item.clone(),
+                                salt,
+                            };
+
+                            let commitment =
+                                Self::get_hashed_commitment(draw.vote.clone(), raw_commitment)?;
+
+                            let raw_vote =
+                                Vote::Revealed { commitment, vote_item: vote_item.clone(), salt };
+                            draw.vote = raw_vote;
+
+                            slashable_amount = draw.slashable;
+                            draw_weight = draw.weight;
+                        } else {
+                            debug_assert!(false, "Binary search should have found a valid index!");
+                        }
+                        (slashable_amount, draw_weight)
```

Comment:
This is a good example of why using `debug_assert!` may not be the best, because you're now forced to return `(slashable_amount, draw_weight)` even though you've already got them stored in the mutable `draws` - just because you don't error if something unexpected happens.